Document/0x05c-Reverse-Engineering-and-Tampering.md

```diff
@@ -1723,96 +1723,89 @@ import android.os.Bundle;
-
-The `onClickListener` implementation for the dialog button doesn't do much:
-
-```java
-package sg.vantagepoint.uncrackable1;
-
-class b implements android.content.DialogInterface$OnClickListener {
-    final sg.vantagepoint.uncrackable1.MainActivity a;
-
-    b(sg.vantagepoint.uncrackable1.MainActivity a0)
-    {
-        this.a = a0;
-        super();
-    }
-
-    public void onClick(android.content.DialogInterface a0, int i)
-    {
-        System.exit(0);
-    }
-}
 ```
 
-It just exits the app. Now intercept it with Frida to prevent the app from exiting after root detection:
+Notice the "Root detected" message in the `onCreate` method and the various methods called in the preceding `if`-statement (which perform the actual root checks). Also note the "This is unacceptable..." message from the first method of the class, `private void a`. Obviously, this method displays the dialog box. There is an `alertDialog.onClickListener` callback set in the `setButton` method call, which closes the application via `System.exit(0)` after successful root detection. With Frida, you can prevent the app from exiting by hooking the MainActivity.a method or the callback inside it. The example below shows how you can hook MainActivity.a and prevent it from ending the application. 
 
-```java
+```javascript
 setImmediate(function() { //prevent timeout
     console.log("[*] Starting script");
 
     Java.perform(function() {
-      bClass = Java.use("sg.vantagepoint.uncrackable1.b");
-      bClass.onClick.implementation = function(v) {
-         console.log("[*] onClick called");
+      var mainActivity = Java.use("sg.vantagepoint.uncrackable1.MainActivity");
+      mainActivity.a.implementation = function(v) {
+         console.log("[*] MainActivity.a called");
       };
-      console.log("[*] onClick handler modified");
+      console.log("[*] MainActivity.a modified");
 
     });
 });
 ```
 
-Wrap your code in the function `setImmediate` to prevent timeouts (you may or may not need to do this), then call `Java.perform` to use Frida's methods for dealing with Java. Afterwards retrieve a wrapper for the class that implements the `OnClickListener` interface and overwrite its `onClick` method. Unlike the original, the new version of `onClick` just writes console output and *doesn't exit the app*. If you inject your version of this method via Frida, the app should not exit when you click the "OK" dialog button.
+Wrap your code in the function `setImmediate` to prevent timeouts (you may or may not need to do this), then call `Java.perform` to use Frida's methods for dealing with Java. Afterwards retrieve a wrapper for `MainActivity` class and overwrite its `a` method. Unlike the original, the new version of `a` just writes console output and *doesn't exit the app*. If you inject your version of this method via Frida, the app should disable the `AlertDialog` entirely or not exit when you click the "OK" dialog button.
```

Comment:
No. Unmodified app exits when the user clicks on OK button on the alert dialog. I wanted to say that you have 2 options if you decide to inject your own frida script. One option is to prevent the alert dialog entirely so the app wont end. The other option is to show the alert dialog, then hook its onclick listener so when when the user clicks on OK button the app does not exit.

The example script I added discards the alert dialog entirely. It is not shown. Current script on the github page hooks the onclick listener method. I thought it is easier to disable the dialog completely so I slightly modified the original approach.

Reply:
Got it. Can you add this to this section?

Reply:
Added some details about the alternative solution and tried to make it more clear.